marumi/ec.beginner/mypage_edit.php

```diff
@@ -0,0 +1,117 @@
+                    <div class="p-signup__selectWrap">
+                    <?php
+                        $year = '';
+                        $month = '';
+                        $day = '';
+                        for ($i=1980; $i <= 2022; $i++) {
+		                    $year .= '<option value="'.$i.'">'.$i.'</option>';
+                        }
+                        for ($i=1; $i <= 12; $i++) {
+                            $month .= '<option value="'.$i.'">'.$i.'</option>';
+                        }
+                        for ($i=1; $i <= 31; $i++) {
+                            $day .= '<option value="'.$i.'">'.$i.'</option>';
+                        }
+
+                        if (isset($currentUser['birthday'])) {
+                            echo "<p class='p-signup__input'>".escape($currentUser['birthday'])."<span class='p-signup__comment'>※変更できません</span></p>";
+                        } else {
+                            echo '
+                                <select class="p-signup__select" name="year">
+                                    <option value="none">--</option>
+                                    '.$year.'
+                                </select><span class="p-select__unit">年</span>
+                                <select class="p-signup__select" name="month">
+                                    <option value="none">--</option>
+                                    '.$month.'
+                                </select><span class="p-select__unit">月</span>
+                                <select class="p-signup__select" name="day">
+                                    <option value="none">--</option>
+                                    '.$day.'
+                                </select><span class="p-select__unit">日</span>
+                            ';
+                        }
+                    ?>
+                    </div>
+
+                    <p class="p-signup__label">性別</p>
+                    <div class="p-signup__radioWrap">
+                        <label class="p-signup__radio">
+                            <input class="p-radio__gender" type="radio" name="gender" value="1" <?php echo $currentUser['gender'] == 1 ? 'checked' : null ?>>
+                            <span>男性</span>
+                        </label>
+                        <label class="p-signup__radio">
+                            <input class="p-radio__gender" type="radio" name="gender" value="2" <?php echo $currentUser['gender'] == 2 ? 'checked' : null ?>>
+                            <span>女性</span>
+                        </label>
+                        <label class="p-signup__radio">
+                            <input class="p-radio__gender" type="radio" name="gender" value="3" <?php echo $currentUser['gender'] == 3 ? 'checked' : null ?>>
+                            <span>無回答</span>
+                        </label>
```

Comment:
ちょっと実際に動かしてないのでこれ動くかわからないのですが、Enumを使うと性別の部分は下記のようにforeachを使ってこんな感じでシンプルに書けるようになるはずです。
これ、実務でよく使うパターンなので覚えとくと良いと思います 👍 
ちなみに PHP8.1以降しか動かないので使うときはMAMPのPHPを8.1にしてください！（バージョンがないかも・・・？）

```suggestion
        <?php foreach (Gender::cases() as $v): ?>
            <label class="p-signup__radio">
                <input class="p-radio__gender" type="radio" name="gender" value="1" <?= $currentUser['gender'] == $v->value ? 'checked' : null ?>>
                <span><?= $v->description() ?></span>
            </label>
        <?php endforeach; ?>
```